### PR TITLE
fix php 8.1 deprecation

### PR DESCRIPTION
**Pull request by alikon**
Pull Request for Issue #114
reported also https://github.com/joomla/joomla-cms/issues/42317

### Summary of Changes
fix deprecated `stripos()` passing null

### CODE REVIEW

File: src/Web/WebClient.php [hunk 1]
```diff
@@ -281,6 +281,7 @@ public function __get($name)
     protected function detectBrowser($userAgent)
     {
         $patternBrowser = '';
+        $userAgent = $userAgent ?? '';
```
Comment (Quy) [inline src/Web/WebClient.php]: Please skip the checks if empty.

**Comment by Chessy65** [comment]
Very thanks. No error text now. I hope, that this come in the new joomla standard version.

File: src/Web/WebClient.php (hunk 1 above)
Comment (alikon) [reply src/Web/WebClient.php]: did yuo mean something like 

```php
if (empty($userAgent)) {
   return;
}
```
?

Comment (Quy) [reply src/Web/WebClient.php]: Maybe like this:

```
if (!empty($userAgent)) {
        $patternBrowser = '';

        // Attempt to detect the browser type.  Obviously we are only worried about major browsers.

        // If we detected a known browser let's attempt to determine the version.
}

// Mark this detection routine as run.
$this->detection['browser'] = true;
```

Comment (alikon) [reply src/Web/WebClient.php]: done

**Comment by richard67** [comment]
@alikon Could you do the same PR here also for the 2.0-dev branch so we could make a release 2.0.5 to fix it in 4.,4-dev of the CMS, too? Let me know if you can do that or if I shall do it. Thanks in advance.

**Comment by alikon** [comment]
i'll try to do it
btw ... 
if i'll run into problems 
i'll ping you 😄

**Comment by alikon** [comment]
see #122 for 2.0

**Pull request merged**